### PR TITLE
Add version string

### PR DESCRIPTION
**Pull request by bigdinotech**
-add version string so that it can be parsed for during firmware upgrade

### CODE REVIEW

**Comment by bigdinotech** [comment]
@SidLeung @eriknyquist please review
This should be the last thing merged before release of bootes

File: src/main.c [hunk 1]
```diff
@@ -48,6 +48,8 @@ char __noinit __stack cdcacm_setup_stack_area[CDCACM_STACKSIZE];
 char __noinit __stack baudrate_reset_stack_area[RESET_STACKSIZE];
 char __noinit __stack usb_serial_stack_area[USBSERIAL_STACKSIZE];
 
+const char *firmware_version="CODK-M_1.6.0";
```
Comment (eriknyquist) [inline src/main.c]: We might need to talk about the format of this version number, shouldn't we be using something like a combination of the BLE version & the actual name of the firmware version, so it matches Git tags? i.e.

BLE-1.5.7_Bootes

(I just made up the BLE version, dunno what it really is)

Comment (eriknyquist) [inline src/main.c]: OK ignore the BLE part. We want to keep a separate BLE firmware version number.

But, I was talking with Sidney, and I think we do want a bit more control over the location of this string.
So, I propose instead to declare an area for this version string in the linker script, at the very end of flash.

Then you can use the linker script symbols in C code to place the version string in the right area, and then the last N bytes of the binary file will be guaranteed to always contain that version string. So no grepping needs to happen, just read a fixed chunk from the very end of the binary file

Comment (eriknyquist) [inline src/main.c]: I can make the linker script changes

Comment (SidLeung) [inline src/main.c]: Would be nice to ping on Martino to let him know that we are tagging the x86 version number at the end of binary image.  Most likely he will have to read a small chunk of data from the flash at an address near the end of file and "grep" for the version string expected.

**Pull request closed without merging**